File: array/1233_remove_sub_folders_from_the_filesystem/1233_remove_sub_folders_from_the_filesystem.go
package main

import (
	"fmt"
	"sort"
	"strings"
)

func removeSubfolders(folder []string) []string {
	sort.Sort(sort.StringSlice(folder))
	out := []string{}
	out = append(out, folder[0])
	t := folder[0]
	for i := 1; i < len(folder); i++ {
		if strings.HasPrefix(folder[i], t+"/") == false {
			out = append(out, folder[i])
			t = folder[i]
		}
	}
	return out
}

func main() {
	folder := []string{"/a", "/a/b", "/c/d", "/c/d/e", "/c/f"}
	res := removeSubfolders(folder)
	fmt.Println(res)
}

File: array/1233_remove_sub_folders_from_the_filesystem/1233_remove_sub_folders_from_the_filesystem.py
from typing import List


class Solution:
    def removeSubfolders(self, folder: List[str]) -> List[str]:
        res, t = [], ' '
        for f in sorted(folder):
            print(f)
            if not f.startswith(t):
                res.append(f)
                t = f + '/'
        return res


folder = ["/a","/a/b","/c/d","/c/d/e","/c/f"]
res = Solution().removeSubfolders(folder)
print(res)

